tck/src/main/java/ee/jakarta/tck/data/framework/junit/extensions/AssertionNameGenerator.java

```diff
@@ -20,36 +20,31 @@
 
 import ee.jakarta.tck.data.framework.junit.anno.Assertion;
 import ee.jakarta.tck.data.framework.junit.anno.Standalone;
+import ee.jakarta.tck.data.framework.utilities.TestProperty;
 
 /**
- * This custom JUnit name generator will append the {@code @Assertion.id}
- * if one exists to the name of the test method at runtime.
+ * <p>This custom JUnit name generator will append the {@code @Assertion.id}
+ * if one exists to the name of the test method at runtime.</p>
  * 
- * This custom JUnit name generator will also append the platform 
- * the test ran on by using the system property jakarta.tck.platform. 
+ * <p>This custom JUnit name generator will also append the platform 
+ * the test ran on by using the {@link ee.jakarta.tck.data.framework.utilities.TestProperty#platform}.</p>
  * 
  * @see ee.jakarta.tck.data.framework.junit.anno.Assertion
  * @see ee.jakarta.tck.data.framework.junit.anno.Standalone
  */
 public class AssertionNameGenerator extends DisplayNameGenerator.Simple implements DisplayNameGenerator {
     
-    /**
-     * Optional property that can be provided by test runners to append the platform to the method name.
-     * This is useful if an test aggregator is being used to run multiple platforms. 
-     * Having the platform name appended to the test will help distinguish the multiple runs.
-     */
-    public static final String platformProperty = "jakarta.tck.platform";
+    private String platform = TestProperty.platform.getValue();
+    private boolean isStandalone = TestProperty.platform.getBoolean();
     
     @Override
     public String generateDisplayNameForClass(Class<?> testClass) {
         //If user provided a platform, then use that
-        String platform = System.getProperty(platformProperty);
```

Comment:
Web and Core are technically profiles, not platforms (they are both part of the Jakarta EE platform) so we should probably rename this at some point for clarity.